Client.Core/Flux/Internal/FluxResultMapper.cs

```diff
@@ -29,164 +29,180 @@
+						{
+							// No need to set field value when column does not exist (default poco field value will be the same)
+							if (recordValues.TryGetValue(col, out var value))
+								SetFieldValue(poco, property, value);
+						}
+					}
+				}
+
+				return poco;
+			}
+			catch (Exception e)
+			{
+				throw new InfluxException(e);
+			}
+		}
+
+
+		/// <summary>
+		/// Maps FluxRecord into custom POCO class.
+		/// </summary>
+		/// <param name="record">the Flux record</param>
+		/// <typeparam name="T">the POCO type</typeparam>
+		/// <returns></returns>
+		/// <exception cref="InfluxException"></exception>
+		internal T ToPoco<T>(FluxRecord record)
+			=> (T)ToPoco(record, typeof(T));
+
+		private void SetFieldValue<T>(T poco, PropertyInfo property, object value)
+		{
+			if (property == null || value == null || !property.CanWrite)
+			{
+				return;
+			}
+
+			try
+			{
+				var propertyType = property.PropertyType;
+
+				//the same type
+				if (propertyType == value.GetType())
+				{
+					property.SetValue(poco, value);
+					return;
+				}
+
+				//handle time primitives
+				if (propertyType == typeof(DateTime))
+				{
+					property.SetValue(poco, ToDateTimeValue(value));
+					return;
+				}
+
+				if (propertyType == typeof(Instant))
+				{
+					property.SetValue(poco, ToInstantValue(value));
+					return;
+				}
+
+				if (value is IConvertible)
+				{
+					// Nullable types cannot be used in type conversion, but we can use Nullable.GetUnderlyingType()
+					// to determine whether the type is nullable and convert to the underlying type instead
+					var targetType = Nullable.GetUnderlyingType(propertyType) ?? propertyType;
+					property.SetValue(poco, Convert.ChangeType(value, targetType));
+				}
+				else
+				{
+					property.SetValue(poco, value);
+				}
+			}
+			catch (InvalidCastException ex)
+			{
+				throw new InfluxException(
+					$"Class '{poco.GetType().Name}' field '{property.Name}' was defined with a different field type and caused an exception. " +
+					$"The correct type is '{value.GetType().Name}' (current field value: '{value}'). Exception: {ex.Message}",
+					ex);
+			}
+		}
+
+		private DateTime ToDateTimeValue(object value)
+		{
+			if (value is DateTime dateTime)
+			{
+				return dateTime;
+			}
+
+			if (value is Instant instant)
+			{
+				return instant.InUtc().ToDateTimeUtc();
+			}
+
+			if (value is IConvertible)
+			{
+				return (DateTime)Convert.ChangeType(value, typeof(DateTime));
+			}
+
+			throw new InvalidCastException(
+				$"Object value of type {value.GetType().Name} cannot be converted to {nameof(DateTime)}");
+		}
+
+		private Instant ToInstantValue(object value)
+		{
+			if (value is Instant instant)
+			{
+				return instant;
+			}
+
+			if (value is DateTime dateTime)
+			{
+				return Instant.FromDateTimeUtc(dateTime);
+			}
+
+			throw new InvalidCastException(
+				$"Object value of type {value.GetType().Name} cannot be converted to {nameof(Instant)}");
+		}
+	}
```

Comment:
Please use `spaces` for indentation. A tab could be a different number of columns depending on your environment, but a space is always one column and the code will be easier to review.